/*
 * This file is part of rasdaman community.
 *
 * Rasdaman community is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Rasdaman community is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with rasdaman community.  If not, see <http://www.gnu.org/licenses/>.
 *
 * Copyright 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014 Peter Baumann / rasdaman GmbH.
 *
 * For more information please see <http://www.rasdaman.org>
 * or contact Peter Baumann via <baumann@rasdaman.com>.
 */

package rasnet;
option java_package="org.rasdaman.rasnet.message";

/**
 * A ServiceRequest is sent by a client to ask the service provider to execute a
 * task and return a response.
 */
message ServiceRequest {
	//This id identifies the request for a particular client
	required uint64 id = 1;

	//The method name uniquely identifies the method the client wants to call.
	required string method_name = 2;

	//The input value represents the serialized request parameter
    required bytes input_value = 3;
}

/**
 * A ServiceResponse is always sent by a service provider in response to a client's request.
 */
message ServiceResponse {
	//The id uniquely identifies the service call for the client.
	//It must be the same as the one in the ServiceRequest message.
	required uint64 id = 1;

	//The output value is the string serialization of the output parameter.
	//This is optional because a ServiceResponse might be an error or a valid result.
    optional bytes output_value = 2;

	//The error component is a human readable string representing an error.
	optional string error = 3;
}
